### ISSUE TITLE
Can't get have_enqueued_email to work in 3.9.0

### ISSUE THREAD

**Issue by pkotnis** [rspec/rspec-rails#2200]
It's probably something trivial, but I can't get the new `have_enqueued_email` matcher to work in 3.9. It looks as if the matcher wasn't available for RSpec.

## What Ruby, Rails and RSpec versions are you using?

Ruby version: ruby 2.6.3p62
Rails version: Rails 6.0.0 (also tried on 5.2.3)
RSpec version: RSpec 3.9 (rspec-core 3.9.0, rspec-expectations 3.9.0, rspec-mocks 3.9.0, rspec-rails 3.9.0, rspec-support 3.9.0)

## Observed behaviour

When running a following super simple test using `have_enqueued_email`

```ruby
RSpec.describe ApplicationMailer, type: :mailer do
  describe '#test_email' do
    it { expect { ApplicationMailer.test_email }.to have_enqueued_email }
  end
end
```

RSpec returns the following message:

```
You must pass an argument rather than a block to `expect` to use the provided matcher (have enqueued email), or the matcher must implement `supports_block_expectations?`.
```

## Expected behaviour

The expected behavior would be for the test to pass (or fail) using a normal message.

## Can you provide an example app?

Here's a fresh Rails 6 app with only one method added in `ApplicationMailer` and a spec for it - https://github.com/pkotnis/have_enqueued_email-test

## Additional input

When I added those two lines to the spec

```ruby
RSpec.describe ApplicationMailer, type: :mailer do
  describe '#test_email' do
    it do
      puts RSpec::Core::Version::STRING
      puts RSpec::Rails::Version::STRING

      expect { ApplicationMailer.test_email }.to have_enqueued_email
    end
  end
end
```

RSpec returned the following error

```
NameError:
       uninitialized constant RSpec::Rails::Version
```

Even tho I can see the `RSpec::Rails::Version::STRING` constant in the source code of the gem.

Thank you for your help in advance.

**Comment by pirj** [comment]
For some reason, `lib/rspec/rails/matchers/have_enqueued_mail.rb` is not loaded, and that results in a [predicate matcher (`have_*`)](https://relishapp.com/rspec/rspec-expectations/docs/built-in-matchers/predicate-matchers) being used instead of `rspec-rails`' provided `have_enqueued_mail`.

I'd suggest to re-generate your config files for the project. Random observations - you require 'rails_helper` manually, while RSpec evaluates `.rspec` file, and requires `spec_helper` as it suggests.

**Comment by JonRowe** [comment]
@pirj The repo is as we setup as we suggest, which is to load `spec_helper` with `.rspec` and require `rails_helper` as needed.

**Comment by JonRowe** [comment]
@pkotnis This is fixed in `4.0.0.beta3` theres a require missing for the matcher

**Comment by pkotnis** [comment]
Thank you @JonRowe <3
